src/core/ack_tracker.c

```diff
@@ -96,6 +96,73 @@ QuicAckTrackerAddPacketNumber(
         !RangeUpdated;
 }
 
+//
+// This function implements the logic defined in Section 6.2 [draft-ietf-quic-frequence-10]
+// to determine if the reordering threshold has been hit.
+//
+BOOLEAN
+QuicAckTrackerDidHitReorderingThreshold(
+    _In_ QUIC_ACK_TRACKER* Tracker,
+    _In_ uint8_t ReorderingThreshold
+    )
+{
+    if (ReorderingThreshold == 0) {
```

Comment:
Actually since we are assuming the smallest tracked packet isnt out of order, we would need atleast 2 subranges for our logic. So it should be  QuicRangeSize(&Tracker->PacketNumbersToAck) < 2